### PR TITLE
Avoid preload of users and threads.

### PR DESCRIPTION
**Pull request by harishkumarv**

### CODE REVIEW

File: store/rules/preLoadThreads.js [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
 		}
 
-		if(changes.entities && user) {
-			Object.keys(changes.entities).forEach(function(key) {
-				if(regex.test(key) && changes.entities[key] && changes.entities[key].room) {
-					query(changes.entities[key].room);
-				}
+		if(future.get("nav").mode === "home") {
+			if (changes.entities && user) {
+				Object.keys(changes.entities).forEach(function(key) {
```
Comment (aravindet) [inline store/rules/preLoadThreads.js]: Change to `for(key in changes.entities)  {`

File: store/rules/preLoadThreads.js [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
+					if (regex.test(key) && changes.entities[key] && changes.entities[key].room && future.get("nav").mode === "home") {
```
Comment (aravindet) [inline store/rules/preLoadThreads.js]: `future.nav.mode === home` has already been tested; unnecessary. Instead, can we also check that the room is not already loaded; i.e. `typeof future.entities[future.entities[key].room] === object` and not null.

File: store/rules/preLoadThreads.js [hunk 3]
```diff
@@ -1,38 +1,40 @@
...
+	store = arguments[2];
 
-	query = require("../bulkQuery.js")(core, store, 'threads');
+	query = require("../bulkQuery.js")(core, store, "threads");
 
 	core.on("setstate", function(changes, next) {
 		var user = store.get("user") || "",
 			room = store.get("nav", room) || "",
-			regex = new RegExp("_" + user + "$");
+			regex = new RegExp("_" + user + "$"),
```
Comment (aravindet) [inline store/rules/preLoadThreads.js]: For completeness, `regex = new RegExp("_" + RegExp.escape(user) + "$")`

We might have to add RegExp.escape to polyfills.

File: store/rules/loadRelatedUsers.js [hunk 4]
```diff
@@ -61,15 +57,51 @@ function loadUsersList(roomId) {
...
+
+
+		if(!changes.entities) {
+			changes.entities = {};
+		}
+		
+
+		if (roomId && roomId !== oldRoomId) {
+			oldRelations.forEach(function(rel) {
+				changes.entities[roomId + "_" + rel.id] = null;
```
Comment (aravindet) [inline store/rules/loadRelatedUsers.js]: Do we really need to do this?

**Comment by aravindet** [comment]
Any update on this @harishkumarv ? Can we merge this for release tomorrow?

**Comment by harishkumarv** [comment]
Yep made the changes needed. Can be merged.

**Pull request merged**